src/inventory_ui.cpp

```diff
@@ -401,14 +454,329 @@ void inventory_selector::print_right_column() const
     }
 }
 
+std::string *inventory_selector::split_name(const std::string name, std::string* rtn) const {
+    rtn[0] = name;
+    rtn[1] = "";
+    if(name.length() > 24) {
+        for(int i = 24; i > 0; i--) {
+            if(name[i] == ' ') {
+                rtn[0] = name.substr(0, i);
+                rtn[1] = name.substr(i + 1);
+                break;
+            }
+        }
+    }
+    return rtn;
+}
+
+void inventory_selector::print_grid(const itemstack_vector &worn, const itemstack_vector &inv, size_t selected) const {
+    
+    nc_color selected_line_color = inCategoryMode ? c_white_red : h_white;
+        
+    int curr_line = 3 - grid_scroll; // Always expected to start at the fourth row
+    int total_height = 0;
+    
+    size_t row_width = (TERMX - 2)/25; // How many cells can fit in a row
+                                    // First two columns are scrollbar, then the leftmost cell border
+                                    // Cells are 25 columns wide, including the right cell border
+    size_t row_i = 0; // How far along the row we are
+    bool cat_continued = false; // Whether or not we wrapped this row
+    bool had_first_cat = false;
+    
+    //player &u = g->u;
+    
+    // First we want to print out worn stuff
+    for(size_t a = 0; a < worn.size(); a++) {
+        const itemstack_or_category &cur_entry = worn[a];
+        if(cur_entry.category == NULL) { // Blank object, skip over
+            continue;
+        }
+        if(cur_entry.it == NULL) { // This is a category header, print out category
+            if(had_first_cat) {
+                curr_line += 6; // Already printed first category, skip 5 lines
+                total_height += 6;
+            } else {
+                had_first_cat = true; // Printing first category, put right at top
+            }
+            const std::string name = cur_entry.category->name;
+            if(curr_line >= 3 && curr_line < TERMY) {
+                center_print(w_inv, curr_line, c_magenta, _("--- %s ---"), name.c_str()); // Center print header
```

Comment:
The comment here is a bit pointless, the called function is named `center_print`, one would assume that it "center prints" its arguments.